### PR TITLE
Go 117 dependency update

### PR DESCRIPTION
**Pull request by LJ-Software**
# Description
Updated the go version of this module to 1.17 and updated the version of all of the module dependencies to latest.

All of the otel modules were updated to newer versions bringing with it many changes in the code to accommodate. Most of the changes are method & attribute name changes in the otel libraries.

Also had to update some of the expected calls in the metrics tests.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| N/A |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to the documentation
- [N/A] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?

- [x] End to End integration test with Observability
```
--- PASS: Test_Metrics (246.54s)
    --- PASS: Test_Metrics/PowerStore_Space_Metrics (192.17s)
    --- PASS: Test_Metrics/PowerStore_FileSystem_Metrics (40.19s)
    --- PASS: Test_Metrics/PowerStore_Volume_Metrics (14.17s)
PASS
ok      karavi-testing/karavi-metrics/metrics-test      246.577s
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,26 +1,88 @@
 module github.com/dell/csm-metrics-powerstore
 
-go 1.14
+go 1.17
```
Comment (AronAtDell) [inline go.mod:3]: Are we coordinating an upgrade of the go version across all CSM components?

**Review by hoppea2 — COMMENTED** [review]
Failure in go unit test action. Do we need to update something on the actions side to 1.17?

/go/pkg/mod/sigs.k8s.io/json@v0.0.0-2021120[82](https://github.com/dell/csm-metrics-powerstore/runs/5057524879?check_suite_focus=true#step:4:82)00746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
/go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211208200746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
note: module requires Go 1.17
FAIL	github.com/dell/csm-metrics-powerstore/internal/common [build failed]
FAIL	github.com/dell/csm-metrics-powerstore/internal/entrypoint [build failed]
FAIL	github.com/dell/csm-metrics-powerstore/internal/k8s [build failed]
FAIL	github.com/dell/csm-metrics-powerstore/internal/service [build failed]
test failed with return code 2

**Comment by alikdell** [comment]
> Failure in go unit test action. Do we need to update something on the actions side to 1.17?
> 
> /go/pkg/mod/sigs.k8s.io/json@v0.0.0-2021120[82](https://github.com/dell/csm-metrics-powerstore/runs/5057524879?check_suite_focus=true#step:4:82)00746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported) /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211208200746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported) note: module requires Go 1.17 FAIL github.com/dell/csm-metrics-powerstore/internal/common [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/entrypoint [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/k8s [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/service [build failed] test failed with return code 2

**Comment by alikdell** [comment]
> > Failure in go unit test action. Do we need to update something on the actions side to 1.17?
> > /go/pkg/mod/sigs.k8s.io/json@v0.0.0-2021120[82](https://github.com/dell/csm-metrics-powerstore/runs/5057524879?check_suite_focus=true#step:4:82)00746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported) /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211208200746-9f7c6b3444d2/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported) note: module requires Go 1.17 FAIL github.com/dell/csm-metrics-powerstore/internal/common [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/entrypoint [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/k8s [build failed] FAIL github.com/dell/csm-metrics-powerstore/internal/service [build failed] test failed with return code 2

Action has go 1.14 hardcoded, one need to upgrade those

**Comment by alikdell** [comment]
actions.yaml has following:
  
  steps:
      - name: Set up Go 1.14+
        uses: actions/setup-go@v2
        with:
          go-version: ^1.14

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by alikdell — APPROVED** [review]

**Review by atye — APPROVED** [review]

**Pull request merged**